modules/compute/main.tf

```
@@ -0,0 +1,55 @@
+resource "azurerm_network_interface" "nic" {
+  name                = "${var.vm_name}-nic"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.subnet_id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.public_ip_id
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "matebox" {
+  name                = var.vm_name
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  size                = var.vm_size
+  admin_username      = "azureuser"
+  network_interface_ids = [
+    azurerm_network_interface.nic.id,
+  ]
+
+  admin_ssh_key {
+    username   = "azureuser"
+    public_key = var.ssh_key_public
+  }
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_virtual_machine_extension" "custom_script" {
+  name                 = "install-app"
+  virtual_machine_id   = azurerm_linux_virtual_machine.matebox.id
+  publisher            = "Microsoft.Azure.Extensions"
+  type                 = "CustomScript"
+  type_handler_version = "2.1"
+
+  settings = <<SETTINGS
+{
+  "fileUris": ["https://raw.githubusercontent.com/LitvinchukRoman/devops_todolist_terraform_task/main/install-app.sh"],
+  "commandToExecute": "bash install-app.sh"
```

The CustomScript extension references `install-app.sh` via a raw GitHub URL (line 51) and executes `bash install-app.sh`. Ensure the referenced script is robust: the repository script currently does not install `git`, uses relative paths, and lacks `set -e`. Update the script to install required packages (`git`), use absolute paths, and fail-fast flags so the extension does not silently fail. See the install-app.sh file for details.